fv3core/stencils/c_sw.py

```diff
@@ -311,174 +302,285 @@ def update_meridional_velocity(
+                func=divergence_corner,
+                externals={
+                    **ax_offsets,
+                },
+                origin=origin,
+                domain=domain,
+            )
+        geo_origin = (self.grid.is_ - 1, self.grid.js - 1, 0)
+        self._geoadjust_ut = FrozenStencil(
+            func=geoadjust_ut,
+            origin=geo_origin,
+            domain=(self.grid.nic + 3, self.grid.njc + 2, self.grid.npz),
+        )
+        self._geoadjust_vt = FrozenStencil(
+            func=geoadjust_vt,
+            origin=geo_origin,
+            domain=(self.grid.nic + 2, self.grid.njc + 3, self.grid.npz),
+        )
+        domain_transportdelp = (self.grid.nic + 2, self.grid.njc + 2, self.grid.npz)
+        ax_offsets_transportdelp = axis_offsets(
+            self.grid, geo_origin, domain_transportdelp
+        )
+        self._transportdelp = FrozenStencil(
+            func=transportdelp,
+            externals={
+                "grid_type": grid_type,
+                **ax_offsets_transportdelp,
+            },
+            origin=geo_origin,
+            domain=(self.grid.nic + 2, self.grid.njc + 2, self.grid.npz),
+        )
+        origin_vort_ke = (self.grid.is_ - 1, self.grid.js - 1, 0)
+        domain_vort_ke = (self.grid.nic + 2, self.grid.njc + 2, self.grid.npz)
+        ax_offsets_vort_ke = axis_offsets(self.grid, origin_vort_ke, domain_vort_ke)
+        self._update_vorticity_and_kinetic_energy = FrozenStencil(
+            func=update_vorticity_and_kinetic_energy,
+            externals={
+                "grid_type": grid_type,
+                **ax_offsets_vort_ke,
+            },
+            origin=origin_vort_ke,
+            domain=domain_vort_ke,
+        )
+        self._circulation_cgrid = FrozenStencil(
+            func=circulation_cgrid,
+            externals={
+                **ax_offsets,
+            },
+            origin=origin,
+            domain=domain,
+        )
+        self._absolute_vorticity = FrozenStencil(
+            func=absolute_vorticity,
+            origin=self.grid.compute_origin(),
+            domain=(self.grid.nic + 1, self.grid.njc + 1, self.grid.npz),
+        )
+
+        domain_meridional = self.grid.domain_shape_compute(add=(0, 1, 0))
+        axis_offsets_meridional = axis_offsets(self.grid, origin, domain_meridional)
+        self._update_y_velocity = FrozenStencil(
+            func=update_y_velocity,
+            externals={
+                "grid_type": grid_type,
+                **axis_offsets_meridional,
+            },
+            origin=origin,
+            domain=domain_meridional,
+        )
+        domain_zonal = self.grid.domain_shape_compute(add=(1, 0, 0))
+        axis_offsets_zonal = axis_offsets(self.grid, origin, domain_zonal)
+        self._update_x_velocity = FrozenStencil(
+            func=update_x_velocity,
+            externals={
+                "grid_type": grid_type,
+                **axis_offsets_zonal,
```

Comment:
This can be changed to an explicit passing of `i_start` and `i_end`.

Reply:
Great suggestion, changed both to pass in only the used externals.